akka-stream/src/main/scala/akka/stream/snapshot/MaterializerState.scala

```diff
@@ -56,7 +56,7 @@ object MaterializerState {
       implicit ec: ExecutionContext): Future[immutable.Seq[StreamSnapshot]] = {
     // Arbitrary timeout: operation should always be quick, when it times out it will be because the materializer stopped
     implicit val timeout: Timeout = 10.seconds
-    (supervisor ? StreamSupervisor.GetChildrenSnapshots).mapTo[StreamSupervisor.ChildrenSnapshots].map(_.seq)
+    (supervisor ? StreamSupervisor.TryGetChildrenSnapshots(3)).mapTo[StreamSupervisor.ChildrenSnapshots].map(_.seq)
```

Comment:
Add a new command to simulate the attempts

Reply:
magic number 3 is not good

Reply:
https://www.youtube.com/watch?v=aU4pyiB-kq0

Reply:
Perhaps we should just fail the future and let users deal with retrying instead. It's not very likely to fail, but can when streams are stopping at the same time this method was called.

Reply:
Yes, the current implementation keeps the original 3 times attempts.

Reply:
I saw @jrudolph comment in the issue as well, which argues in the other direction, make sure we complete with partial results etc. I'm not quite sure what is the best behavior in face of rapidly changing sets of streams, which I think is the only case when it'd fail. Well that and an overloaded system that does not have cycles to respond in time.

Reply:
In general, in face of rapidly changing sets of streams, there's no right or wrong about what to include because of the inherent race conditions.

The main use case where it is used right now is at the end of tests where it usually fails because some streams might just be terminating. If retries are only there to help with this case, maybe we could do better even without retries?

It could work like this:

 * when the snapshots is requested, the current set of children is the one we care about, for all these we request a snapshot, noting the original ActorRef
 * when some snapshots fail to be reported, we can check after the fact:
   * a snapshot might be missing due to a terminated child (check ActorRef vs. `context.children` when results are in or after timeout), in that case we can just ignore the timeout for that entry
   * a snapshot is missing but the ActorRef is still alive: report as alive but without data, in that case `assertAllStagesStopped` should fail

Would that work?

Reply:
Sounds good to me

Reply:
@jrudolph Did you see any error of the original issue or just looked at the code and find it?

Reply:
I can update later.

Reply:
@jrudolph I have updated the fix